security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -846,22 +846,7 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
 		return nil, err
 	}
 
-	// If token is jwt format, construct host name from jwt with format like spiffe://cluster.local/ns/foo/sa/sleep
-	// otherwise just use sdsrequest.resourceName as csr host name.
-	csrHostName := connKey.ResourceName
-	// TODO (liminw): This is probably not needed. CA is using claims in the credential to decide the identity in the certificate,
-	// instead of using host name in CSR. We can clean it up later.
-	if sc.secOpts.CredFetcher == nil {
-		csrHostName, err = constructCSRHostName(sc.configOptions.TrustDomain, token)
-		if err != nil {
-			cacheLog.Warnf("%s failed to extract host name from jwt: %v, fallback to SDS request"+
-				" resource name: %s", logPrefix, err, connKey.ResourceName)
-			csrHostName = connKey.ResourceName
-		}
-	}
-	cacheLog.Debugf("constructed host name for CSR: %s", csrHostName)
 	options := pkiutil.CertOptions{
-		Host:       csrHostName,
```

Comment:
I am not sure if we can completely remove the field from CSR, or should fill it with a dummy value (like we do for GCE VM case). John, can you verify that istiod is issuing certificate correctly with this change? @JimmyCYJ @myidpt Do we have any integration tests covering this code path?

Reply:
@JimmyCYJ @myidpt?

Reply:
I asked @JimmyCYJ. We probably only have unit test in place. John, could you manually verify that with the change, istiod is still issuing certificate correctly?

Reply:
What am I supposed to verify? Of course mTLS between pods works, our testing isn't so bad that we don't test that